src/Node/ForNode.php

```diff
@@ -101,7 +101,11 @@ public function compile(Compiler $compiler): void
         $compiler->write("\$_parent = \$context['_parent'];\n");
 
         // remove some "private" loop variables (needed for nested loops)
-        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\'], $context[\'loop\']);'."\n");
+        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\']');
+        if ($this->getAttribute('with_loop')) {
+            $compiler->write(', $context[\'loop\']');
```

Comment:
this needs to use `->raw` so that it does not add indentation again (as it is not meant to write new lines but to continue the content of the current line).
Same below for the end of the line

Reply:
Good to know, changed!